app/components/goodcity/item-search-overlay.js

```diff
@@ -7,32 +7,66 @@ export default Ember.Component.extend(SearchMixin, {
   autoLoad: true,
   store: Ember.inject.service(),
   perPage: 10,
+  packageService: Ember.inject.service(),
+  messageBox: Ember.inject.service(),
+  i18n: Ember.inject.service(),
 
   init() {
     this._super("item-search-overlay");
     this.set("displayResults", true);
   },
 
+  storageTypeName: Ember.computed.alias("entity.storageTypeName"),
+
   hasSearchText: Ember.computed("searchText", function() {
     return !!this.get("searchText");
   }),
 
+  addSingleton(item) {
+    let itemLocations = item.get("locations");
+    const params = {
+      item_id: item.id,
+      task: "pack",
+      location_id: itemLocations.get("lastObject").id,
+      quantity: item.get("onHandQuantity")
+    };
+
+    this.get("packageService")
+      .addRemoveItem(this.get("entity.id"), params)
+      .then(data => {
+        this.sendAction("onSingletonAdd");
+        this.set("open", false);
+        this.destroy();
+      })
+      .catch(response => {
+        let error_message =
+          response.responseJSON && response.responseJSON.errors[0];
+        this.get("messageBox").alert(
+          error_message || this.get("i18n").t("unexpected_error")
+        );
+      });
+  },
+
   actions: {
     cancel() {
       this.set("searchText", "");
       this.set("open", false);
     },
 
     selectItem(item) {
-      this.set("open", false);
+      if (item.get("onHandQuantity") === 1) {
+        return this.addSingleton(item);
```

Comment:
Why `return` ?

Reply:
good catch, return not required, will remove it.

Reply:
@abulsayyad123 , return required because I don't want to proceed further if I get inside this block.